### PR TITLE
Add new endpoint to get players by name

### PR DESCRIPTION
**Pull request by FernandoRoldan93**
Added new endpoint as requested in issue #4.

### CODE REVIEW

File: app/main.py [hunk 1]
```diff
@@ -26,10 +26,15 @@ async def ajo_stream(count: int = 50, id: str = 0):
...
-async def get_player(player_id):
+async def get_player(player_id: int):
     ajos = await r.zscore("lb", player_id)
     return {"player_id": ajos}
 
+@app.get("/player/{player_name}")
+async def get_player_by_name(player_name: str):
+    ajos = await r.zscore("lb", player_name)
+    return {"player_name": ajos}
+
```
Comment (axl89) [inline app/main.py:37]: This won't work. The `lb` data structure only contains IDs. See: https://github.com/ass-a-service/ajobot/blob/master/ARCHITECTURE.md

Comment (axl89) [reply app/main.py:37]: @Zymna what do you suggest we could do here? Given a `player_name` what would be the best way to obtain its discord user ID?

Comment (Zymna) [reply app/main.py:37]: It seems possible for two users to have the same name, hence we need N reverse lookup keys.
Currently the id/name relationship is done when an ajo is gained `ajo.lua`, we suggest moving this out into the event processing (`on_ajo`) and adding 1 set per name.

To simplify, the check if name changed should be in python (`on_ajo`):
```py
def on_ajo(id, name):
    current_name = r.get(id)
    if current_name != name:
        return r.evalsha("name_change", ...)
```

The name change process should be in lua:
```lua
-- name_change.lua
local name_key = KEYS[1] -- the name key is the id
local reverse_key = KEYS[2] -- the lookup key is the name
local prev_reverse_key = KEYS[3] -- previous name lookup key

local id = ARGV[1]
local name = ARGV[2]
local prev_name = ARGV[3]

-- replace with new name, cleanup old set and update new one
redis.call("set", name_key, name)
redis.call("srem", prev_reverse_key, id)
redis.call("sadd", reverse_key, id)

return "OK"
```

Name should include discriminator.

Comment (Zymna) [reply app/main.py:37]: https://github.com/ass-a-service/ajobot/pull/94 now stores the identifier relation with user name in a hash named `ids` . You can retrieve a user's identifier with `hget ids {name}#{discriminator}`

**Pull request closed without merging**